packages/core/src/common/utils/domUtils.ts

```diff
@@ -150,6 +150,10 @@ function throttleImpl<T extends Function>(
 }
 
 export function clickElementOnKeyPress(keys: string[]) {
-    return (e: React.KeyboardEvent) =>
-        keys.some(key => e.key === key) && e.target.dispatchEvent(new MouseEvent("click", { ...e, view: undefined }));
+    return (e: React.KeyboardEvent) => {
+        if (keys.some(key => e.key === key)) {
+            e.preventDefault(); // Prevent spacebar from scrolling the page
```

Comment:
this could possibly cause issues if a MenuItem has a text input inside it, I think. it's not really recommended, but I wouldn't be surprised if something like this exists in the wild:

https://codesandbox.io/p/devbox/distracted-cookies-sv5mlx

![image](https://github.com/palantir/blueprint/assets/723999/a82fbbfb-384e-4056-acdd-1cde79864dad)


we should test / verify this behavior in the docs by adding an item which contains an `<InputGroup>` to the [Menu docs example](https://blueprintjs.com/docs/#core/components/menu.menu-item)

one idea for preventing a regression here is to check if the keypress occurred in a text input, using another utility fn in this same module:

```ts
if (elementIsTextInput(e.target as HTMLElement)) {
  // don't prevent default
}
```

Reply:
Ah, I didn't consider someone could be nesting a text element in a `MenuItem`. I can add a check and docs example for that.

Though if that were the case today, it would be issuing a click on the `MenuItem` any time they hit the spacebar. I assume we'd want to keep that behavior to avoid any breaks?

Reply:
> it would be issuing a click on the MenuItem any time they hit the spacebar

Yeah I suppose so, but they might be doing their own `preventDefault` to work around this. Agree that we should preserve the existing behavior as much as possible to avoid surprises